consensus/istanbul/backend/announce.go

```diff
@@ -213,16 +224,24 @@ func (sb *Backend) announceThread() {
 				// Regardless, send the queryEnode so that it will at least be
 				// processed by this node's peers. This is especially helpful when a network
 				// is first starting up.
-				if err := sb.generateAndGossipQueryEnode(announceVersion, queryEnodeFrequencyState == LowFreqState); err != nil {
+				if err := sb.generateAndGossipQueryEnode(sb.GetAnnounceVersion(), queryEnodeFrequencyState == LowFreqState); err != nil {
 					logger.Warn("Error in generating and gossiping queryEnode", "err", err)
 				}
 			}
 
 		case <-sb.updateAnnounceVersionCh:
+			// Drain this channel, as the update version action will address all requests.
+
+		drainLoop:
```

Comment:
You do not need a drain loop if this channel has size 1 and it is sent to with non-blocking send. https://github.com/celo-org/celo-blockchain/pull/1125/files#diff-49675a1e6ab38a575ed8c1e9a5f31145L230-L240